### ISSUE TITLE
Failed to install provider due to expired OpenPGP key

### ISSUE THREAD

**Issue by DT-NIKOLAR** [iSchluff/terraform-provider-keepass#30]
## Description: 
I encountered an error while running terraform init with the ischluff/keepass provider v1.0.0. The error message indicates that the installation failed due to an expired OpenPGP key. 

```
╷
│ Error: Failed to install provider
│
│ Error while installing ischluff/keepass v1.0.0: error checking signature:
│ openpgp: key expired
╵
```

## Steps to Reproduce:
1. Run ```terraform init``` with the following configuration
```
provider "keepass" {
  database = "***"
  password = ***
}
```

Terraform Version: v1.6.0
Operating System: Linux
Provider Version: ischluff/keepass v1.0.0

Could you please update the OpenPGP key or provide guidance on how to resolve this issue?

**Comment by iSchluff** [comment]
Please try again, I have extended the signing key

**Comment by DT-NIKOLAR** [comment]
I tried running terraform init with version 1.0.1, but I am still getting: 
```
│ Error: Failed to install provider
│
│ Error while installing ischluff/keepass v1.0.1: error checking signature:
│ openpgp: key expired
```

**Comment by iSchluff** [comment]
Ok this hole was deeper than I expected.
Apparently the checking of signing key expirations was an unintended change by the terraform developers. See https://github.com/hashicorp/terraform/issues/33984

You are probably using an affected terraform version and may fix this by upgrading to atleast terraform-1.6.1 or downgrading to 1.5.7

Concerning the expired key, I have published an updated version to the ubuntu key servers.
Apparently this check is run against the gpg public key provided by the terraform registry. I have no way to actually update the key used by the registry, I just hope that they eventually refresh the keys they use there.

**Comment by DT-NIKOLAR** [comment]
Thank you for the research and help. I managed to download the new 1.0.1 version with terraform 1.6.1. 

In our case, we made the decision to transition to locally installed providers. For those encountering challenges with locally installed providers and Terraform/Terragrunt, here are some key points to consider:

* Terraform writes provider information in the state file. Even if you remove any registry references to a particular provider from your configuration files, you will still need to do a [terraform state replace-provider](https://developer.hashicorp.com/terraform/cli/commands/state/replace-provider). 

* You will have to run terragrunt init first to install the local provider, before failing on the PGP Key check for the remote provider. The _state replace-provider_ cmd needs both providers to be initialized for the specific module. 

* Since terragrunt supports inter-module dependencies, running terragrunt init within a dependent module failed to initialize the dependency module's local providers. The solution was to run a terragrunt run-all init --terragrunt-include-external-dependencies for it to be able to install the local providers. 

Hope the above helps someone ;)
